src/unstract/sdk/adapters/x2text/llm_whisperer_v2/src/helper.py

```diff
@@ -352,13 +358,13 @@ def write_output_to_file(output_json: dict, output_file_path: Path) -> None:
             metadata_file_name = output_file_path.with_suffix(".json").name
             metadata_file_path = metadata_dir / metadata_file_name
             # Ensure the metadata directory exists
-            metadata_dir.mkdir(parents=True, exist_ok=True)
+            fs.mkdir(create_parents=True, path=metadata_dir)
             # Remove the "result_text" key from the metadata
             metadata = {
                 key: value for key, value in output_json.items() if key != "result_text"
             }
             metadata_json = json.dumps(metadata, ensure_ascii=False, indent=4)
             logger.info(f"Writing metadata to {metadata_file_path}")
-            metadata_file_path.write_text(metadata_json, encoding="utf-8")
```

Comment:
@harini-venkataraman with the new `fs` option added - how do we distinguish the writing to `output_file_path` and `metadata_file_path`?

Reply:
We have a `path` param, which will distinguish between the path in the initialized provider of fs instance.